File: LeetCode_203/Main.py
from typing import Optional

# Definition for singly-linked list.
class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next

class Solution:
    def removeElements(self, head: Optional[ListNode], val: int) -> Optional[ListNode]:
        resultHead = current = None

        while head:
            if head.val != val:
                if resultHead:
                    current.next = head
                    current = current.next
                else:
                    resultHead = head
                    current = head
            head = head.next
        
        if resultHead:
            current.next = None
        
        return resultHead

File: LeetCode_203/Main.cpp
#include<bits/stdc++.h>
using namespace std;


// Definition for singly-linked list.
struct ListNode {
    int val;
    ListNode *next;
    ListNode() : val(0), next(nullptr) {}
    ListNode(int x) : val(x), next(nullptr) {}
    ListNode(int x, ListNode *next) : val(x), next(next) {}
};

class Solution {
public:
    ListNode* removeElements(ListNode* head, int val) {
        ListNode* resultHead = nullptr;
        ListNode* current = nullptr;

        while (head) {
            if (head->val != val) {
                if (resultHead ==  nullptr) {
                    resultHead = head;
                    current = head;
                } else {
                    current->next = head;
                    current = current->next;
                }
            }
            head = head->next;
        }

        if (resultHead) {
            current->next = nullptr;
        }

        return resultHead;
    }
};